### PR TITLE
Quick note about CSS in JS

### PR DESCRIPTION
**Pull request by trankhacvy**
#### What's this PR do?
- [x] Add quick note about **CSS in JS**

### CODE REVIEW

File: Engineering/Frontend/CSS in JS.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+tags: frontend, css-in-js
+author: Tran khac Vy
+---
+
+**CSS-in-JS** is a styling technique where Javascript is used to style the component. These are multiple implementation of this concept in the form of libraries such as [emotion](https://emotion.sh), [styled-component](https://styled-components.com/), [JSS](https://cssinjs.org). They airm to tacke the limitations of CSS, such as lack of dynamic functionality, scoping, and portability.
+
+## Benefits of using CSS-in-JS
+
+### Scoping
+- With CSS-in-JS, writing new styles cannot affect anything else in other places on the site, so there’s no need to worry about writing a style that has bad or unintended consequences elsewhere due to a selector in the global scope.
```
Comment (zlatanpham) [inline Engineering/Frontend/CSS in JS.md:11]: For one item list, it'd be better to convert it into a paragraph.

```suggestion
### Scoping
With CSS-in-JS, writing new styles cannot affect anything else in other places on the site, so there’s no need to worry about writing a style that has bad or unintended consequences elsewhere due to a selector in the global scope.
```

File: Engineering/Frontend/CSS in JS.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## Disadvantages of CSS-in-JS
+
+### Runtime cost
+- When CSS is generated from JavaScript at runtime, in the browser, there is an inherent overhead.
+
+### Learning curve
+- CSS-in-JS definitely has a learning curve, especially if you have used neither component-based frameworks nor web components before. Besides learning the new syntax, you also need to pick up a new way of thinking, which needs time and might slow down your development workflow for a while.
+
+### Unreadable class names
+- Automatically generated selectors significantly worsen code readability. This can be a huge concern for you if you regularly use your browser’s developer tools for debugging.
```
Comment (zlatanpham) [inline Engineering/Frontend/CSS in JS.md:34]: I think there is still a config to have the readable version in dev mode. See https://styled-components.com/docs/tooling#better-debugging

File: Engineering/Frontend/CSS in JS.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+### Dynamic Functionality
+- As CSS-in-JS is essentially JavaScript code, you can apply complex logic to your style rules, such as loops, conditionals, variables, state-based styling, and more.
+
+### Dead code elimination
+- CSS-in-JS helps with removing dead code. The only styles that are loaded are the styles for the components in use at any given time. There’s no shipping any unused styles. When a component dies, so does its styles.
+
+### Developer ergonomics
+- It can be nice to have styles in the same file (or otherwise very close to) the component itself. In the same way, some developers feel very comfortable in JSX. Also being able to style things without any scoping worry means developers may feel empowered about styling rather than intimidated by it.
+
+## Disadvantages of CSS-in-JS
```
Comment (zlatanpham) [inline Engineering/Frontend/CSS in JS.md:25]: I think adding an extra bundle size might be worth mentioning?

File: Engineering/Frontend/CSS in JS.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+### Runtime cost
+- When CSS is generated from JavaScript at runtime, in the browser, there is an inherent overhead.
```
Comment (zlatanpham) [inline Engineering/Frontend/CSS in JS.md:28]: Some tools like https://github.com/callstack/linaria have the solution for the problem but certainly they have its own drawbacks

**Review by zlatanpham — COMMENTED** [review]
Looks good. I've left a couple of suggestions. Let me know if you have any questions.

File: Engineering/Frontend/CSS in JS.md (hunk 2 above)
Comment (trankhacvy) [reply Engineering/Frontend/CSS in JS.md:35]: Just add a sentence to make this opinion clearer

> - Currently, many CSS-in-JS libraries try to provide meaningful class names based on the declaration name or component name in development mode. Some of them even let you customize the class name generator function. In production mode though, the class names still hard to read and debug.

File: Engineering/Frontend/CSS in JS.md (hunk 4 above)
Comment (trankhacvy) [reply Engineering/Frontend/CSS in JS.md:28]: Just add a sentence to make this opinion clearer

> - Some CSS-in-JS libraries try to overcome this overhead by extracting CSS files during the build time (like [linaria](https://linaria.dev)) but it comes with other [trade-offs](https://github.com/styled-components/styled-components/issues/2377)

File: Engineering/Frontend/CSS in JS.md (hunk 3 above)
Comment (trankhacvy) [reply Engineering/Frontend/CSS in JS.md:25]: Added. `Adding library => more code => increase bundle size.`
Do you have any suggestions for a more meaningful full sentence?

**Pull request merged**